### PR TITLE
bench/bin: Support simple command line args

### PR DESCRIPTION
**Pull request by MyonKeminta**
Codes under `benches/bin` are compiles as bin file `bench-tikv`. Due to that there are more than one benches in this bin file, it is useful to support specifying what bench to run by passing command line args. Now we can do `./bench-tikv bench1 bench2 ...` or `cargo run --bin bench-tikv --release -- bench1 bench2 ...` to run specified benches in it. If run without command line argument, it will run all benches by default.

### CODE REVIEW

File: benches/bin/bench-tikv.rs [hunk 1]
```diff
@@ -87,6 +87,8 @@ fn print_result(smp: BenchSamples) {
     println!("{}", test::fmt_bench_samples(&smp));
 }
 
+use std::env;
```
Comment (siddontang) [inline benches/bin/bench-tikv.rs]: move this upward to the place below the crate declaration.

File: benches/bin/bench-tikv.rs [hunk 2]
```diff
@@ -96,6 +98,37 @@ fn main() {
         );
     }
     // TODO allow user to specify flag to just bench some cases.
-    raftstore::bench_raftstore();
-    mvcc::bench_engine();
+
+
+    let mut args: Vec<_> = env::args().skip(1).collect();
+
+    let available_options = vec!["raftstore", "tombstone-scan"];
```
Comment (siddontang) [inline benches/bin/bench-tikv.rs]: why rename to "tombstone-scan"? maybe "mvcc" here?

File: benches/bin/bench-tikv.rs [hunk 3]
```diff
@@ -96,6 +98,37 @@ fn main() {
...
+
+    if args.is_empty() {
+        args = available_options.iter().map(|&s| String::from(s)).collect();
+    }
+
+    if args[0] == "-h" || args[0] == "--help" {
```
Comment (siddontang) [inline benches/bin/bench-tikv.rs]: why not using clap to simplify the arg operation?

File: benches/bin/bench-tikv.rs (hunk 2 above)
Comment (MyonKeminta) [reply benches/bin/bench-tikv.rs]: It does bench "tombstone-scan". In `benches/bin/mvcc.rs`. I don't know why either.

Comment (siddontang) [reply benches/bin/bench-tikv.rs]: this is still included in the mvcc.

Here I think you should use mvcc.

**Comment by siddontang** [comment]
any update? @MyonKeminta

**Comment by MyonKeminta** [comment]
Sorry but I think this can be closed..

**Comment by siddontang** [comment]
I think we need to use flag to benchmark different cases, so it is worth to go on.

File: benches/bin/bench-tikv.rs [hunk 4]
```diff
@@ -94,7 +96,30 @@ fn main() {
...
+        .group(ArgGroup::with_name("benches").args(&available_benches))
+        .get_matches();
+
+    let benches: Vec<_> = if let Some(args) = matches.values_of("benches") {
+        args.collect()
+    } else {
+        available_benches.to_vec()
+    };
+
+    println!("{:?}", benches);
```
Comment (zhangjinpeng1987) [inline benches/bin/bench-tikv.rs]: Begin to run {:?} benches.

**Comment by siddontang** [comment]
Look better. Can you show the example of how to use it now?

**Comment by MyonKeminta** [comment]
Now benches in bin `bench-tikv` can choose to run some of the benches in it. If no argument is provided, it will run all benches in it.
For example:
```bash
./bench-tikv
```
or:
```bash
cargo run --release --bin bench-tikv
```
It will run all benches.
```bash
./bench-tikv raftstore
```
or
```bash
cargo run --release --bin bench-tikv -- raftstore
```
It will bench `raftstore` only.
you can also pass multiple args to run multiple items.

**Comment by siddontang** [comment]
LGTM

**Pull request merged**